go/glome/glome.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package glome implements GLOME protocol.
+package glome
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/curve25519"
+	"io"
+)
+
+const (
+	// PrivateKeySize is the size of a PrivateKey in bytes.
+	PrivateKeySize = 32
+	// PublicKeySize is the size of a PublicKey in bytes.
+	PublicKeySize = 32
+	// SharedSecretSize is the size of a SharedSecret in bytes.
+	SharedSecretSize = 32
+	// Maximum size allow for a Tag
+	MaxTagSize = 32
+)
+
+var (
+	errInvalidPublicKey = errors.New(
+		fmt.Sprintf("glome: invalid public key - byte slice len is not %d", PublicKeySize))
+	errInvalidPrivateKey = errors.New(
+		fmt.Sprintf("glome: invalid private key - byte slice len is not %d", PrivateKeySize))
+	errInvalidTagSize = errors.New(
+		fmt.Sprintf("glome: invalid tag size - MinTagSize must be in range 1-%d", MaxTagSize))
```

Comment:
`1` is a constant too, so you may, for example, save it as `MinPossibleTagSize`.

Reply:
Fixed.